docs/design/paginators.md

```diff
@@ -70,127 +69,113 @@ used in a model.
 
-fun<T> SdkAsyncIterable<Iterable<T>>.flatten(): SdkAsyncIterable<T> = TODO("not-implemented")
-fun <T> SdkAsyncIterable<T?>.filterNotNull(): SdkAsyncIterable<T> = TODO("not-implemented")
+### Pagination over Nested Item
 
-suspend fun<T> SdkAsyncIterable<T>.toList(): List<T> = TODO("not-implemented")
-suspend fun<T> SdkAsyncIterable<T>.toSet(): Set<T> = TODO("not-implemented")
+If `items` is specified in the model for a given operation, then a `Flow` transform function (extending the return type of it's parent) will be generated that 
+allows the nested item to be paginated over by producing another flow that works from the base response flow.
+The targeted type will provide iteration to the `item` element for the user automatically.
 
-/**
- * Terminal operator that collects the given iterable and calls [action] for each result
- */
-inline fun <T> SdkAsyncIterable<T>.collect(crossinline action: suspend (T) -> Unit):Unit = TODO()
+Here is an example that follows from the previous example (codegen):
+
+```kotlin
+fun Flow<ListFunctionsResponse>.onFunctionConfiguration(): Flow<FunctionConfiguration> =
+   transform() { response ->
+      response.functions?.forEach {
+         emit(it)
+      }
+   }
 ```
 
-Pagination by default will always generate an `SdkAsyncIterable` over the normal operation output type. 
-If `items` is specified then the generated paginator will be “specialized” to have an additional field that iterates 
-only over the member targeted by `items`. The targeted type will be flattened for the user automatically.
 
-An example of this is demonstrated below:
+### Creating a Paginator
 
+Each operation that supports pagination would receive an extension function (based from the service client interface) to create a paginator. 
+The function generated always begins with the prefix `paginate`, helping with the `discoverablity` design consideration, by allowing users to quickly 
+view all paginatable operations by specifing this literal prefix in the IDE.  In a similar way, extension functions that provide direct access to a paginator's
+nested member will always begin with the prefix `on`.  Once customers are familiar with these two prefixes, discovering what pagination options are available for a given
+client or paginated response is simple and consistent.
 
-```kotlin
-class ListFunctionsPaginator(
-    private val client: LambdaClient,
-    private val initialRequest: ListFunctionsRequest
-): SdkAsyncIterable<ListFunctionsResponse>, SdkAsyncIterator<ListFunctionsResponse> {
-   private var cursor: String? = null
-   private var isFirstPage: Boolean = true
+Using the model in the introduction would produce the following:
 
-   override operator fun iterator(): SdkAsyncIterator<ListFunctionsResponse> = this
+### Examples
 
-   override suspend operator fun hasNext(): Boolean
-           = isFirstPage || (cursor?.isNotEmpty() ?: false)
+#### Usage - Manual Pagination
```

Comment:
nit: this header makes no sense anymore and could just be renamed to simply `Usage`. Manual pagination referred to driving a paginator class from the previous design